clickhouse/base/socket.cpp

```diff
@@ -151,6 +151,19 @@ SocketHolder::operator SOCKET () const noexcept {
     return handle_;
 }
 
+void SocketHolder::SetTcpKeepAlive(int idle, int intvl, int cnt) noexcept {
+    int val = 1;
+    setsockopt(handle_, SOL_SOCKET, SO_KEEPALIVE, &val, sizeof val);
+
+#if defined(_linux_)
+    setsockopt(handle_, IPPROTO_TCP, TCP_KEEPIDLE, &idle, sizeof idle);
```

Comment:
Please add brackets to "sizeof" operator in every case. Example - "sizeof(idle)"